### PR TITLE
Source drift

### PR DESCRIPTION
**Pull request by eugene-kulak**
## Contributor Checklist
Thanks for contributing to Airbyte! Please complete the following items in order so we can review your PR.
- [x] Followed all the instructions in the locally generated checklist and your connector is functional & ready for review
- [x] Ran the standard test suite locally via `./gradlew :airbyte-integrations:connectors:source-<your_source_name>:standardSourceTestPython` and pasted the summarized output as a comment in this PR

## Reviewer Pre-merge Checklist 
- [x] Finished iterating with the PR author on the code*
- [x] Created a branch off master to merge this PR into*
- [x] Inject the credentials in CI via `./tools/integrations/ci_credentials.sh` and `.github/workflows/test-command.yml`*
- [x] Added the credentials for this integration to Github secrets 
- [x] Run standard tests on this branch by commenting `/test connector=<name>`*
- [x] Add entry in `airbyte-config/init/src/main/resources/seed/source_definitions.yaml` to use the new source in Airbyte core
- [ ] Deployed the connector to Dockerhub via `./tools/integrations/manage.sh publish airbyte-integrations/connectors/source-<name>`

## Documentation
- [x] Add docs in `docs/integrations/sources/` folder in line with the documentation template found in `docs/contributing-to-airbyte/templates/integration-documentation-template.md`.
- [x] Add link to create docs file to `docs/SUMMARY.md`
- [x] Include a link to the documentation in the `README.md`

### CODE REVIEW

**Comment by eugene-kulak** [comment]
```
Test run finished after 23826 ms
[         2 containers found      ]
[         0 containers skipped    ]
[         2 containers started    ]
[         0 containers aborted    ]
[         2 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         0 tests failed          ]

```

**Review by sherifnada — COMMENTED** [review]
question about which streams we're syncing, remaining is style or clarification. 

Does this API support incremental sync? For the record, in the coming weeks we'll slow down new connector development and add features like incremental or syncing child streams to our current connectors. Would the current structure lend itself well to that?

File: airbyte-integrations/connectors/source-drift/source_drift/client/client.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from .api import APIClient
+from .common import AuthError, ValidationError
+
+
+class Client(BaseClient):
+    def __init__(self, access_token: str):
+        super().__init__()
+        self._client = APIClient(access_token)
+
+    def stream__accounts(self) -> Iterator[dict]:
```
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-drift/source_drift/client/client.py:38]: `Contacts` doesn't provide a list method, only get
`Messages` is a child entity and needs conversation id to be fetched.
In the future, we might solve the problem with fetching child streams, so I keep it here.

File: airbyte-integrations/connectors/source-drift/source_drift/client/api.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    @_parsed_response
+    def patch(self, url, data, **kwargs):
+        return requests.patch(self.full_url(url), data=json.dumps(data), headers=self._headers, **kwargs)
+
+    @_parsed_response
+    def get(self, url, **kwargs):
+        return requests.get(self.full_url(url), headers=self._headers, **kwargs)
+
+    @_parsed_response
+    def delete(self, url, **kwargs):
```
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-drift/source_drift/client/api.py:67]: sure, will do, IMO access should be restricted with Access Token permissions

**Comment by eugene-kulak** [comment]
For incremental sync I see the following way it might land:
```python
class BaseClient:
...
 def set_state(*args, **kwargs):  # the actual state object is TBD
    pass
 def get_state(*args, **kwargs):
    pass
    
class BaseSource:
....
    def read(..., stream...):
       client = ...
       if support_state:
           client = set_state(read_state_object(stream))
       yield from save_state(client.read_stream(), every=1000)
```

File: airbyte-integrations/connectors/source-drift/sample_files/configured_catalog.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
+{
+  "streams": [
+    {
+      "stream": {
+        "name": "accounts",
```
[the remark this answers is not in the document]
Comment (sherifnada) [reply airbyte-integrations/connectors/source-drift/sample_files/configured_catalog.json:5]: @eugene-kulak can merge once this is addressed

File: airbyte-integrations/connectors/source-drift/source_drift/client/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sherifnada) [reply airbyte-integrations/connectors/source-drift/source_drift/client/client.py:38]: sgtm

**Review by sherifnada — APPROVED** [review]
modulo adding last stream to configured catalog for testing

File: airbyte-integrations/connectors/source-drift/sample_files/configured_catalog.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-drift/sample_files/configured_catalog.json:5]: @sherifnada yes, sorry,  I have added the last stream

File: airbyte-integrations/connectors/source-drift/source_drift/client/api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-drift/source_drift/client/api.py]: I have dropped delete and added the comment to PATCH method, POST used in the  tocken verification

**Comment by sherifnada** [comment]
Merged in #1393 . Thanks @eugene-kulak !! 🎉

**Pull request closed without merging**